en/guide/toolchain.md

```diff
@@ -0,0 +1,121 @@
+```cmake
+if(NOT MSVC)
+    add_definitions("-std=c++11 -Wall -Wextra -Werror")
+else()
+    add_definitions("-std=c++11 -WX -W2")
+    set(platform_libs "Ws2_32.lib")
+endif()
+
+# Add DEBUG define for Debug target
+set(CMAKE_CXX_FLAGS_DEBUG "-DDEBUG")
+```
+
+Add the line below to find a thread library on Linux, Mac, and Windows. This is required in order to use DroneCore!
+```cmake
+find_package(Threads REQUIRED)
+```
+
+The `include_directories()` command tells *cmake* where to find locally installed header files (if DroneCore is installed globally this is not required). 
+
+```cmake
+include_directories(
+    # Not needed if installed system-wide
+    ${CMAKE_SOURCE_DIR}/../../install/include
+)
+```
+
+> **Note** The fragment above is taken from one of the DroneCore examples, and specifies a relative path from an example sub-directory to where DroneCore exports the header files by default. 
+> If you have installed DroneCore locally and your application is not located alongside the DroneCore examples you will need to change this. 
+>
+>  Note that `CMAKE_SOURCE_DIR` is the directory in which **CMakeLists.txt** is stored, and provides a useful way of defining relative paths to the header files.
+
+Then use `add_executable()` to specify the name for your generated executable and your source files. 
+```cmake
+add_executable(your_executable_name
+    your_source_file.cpp
+)
+```
+
+The final sections of the file defines the libraries to link against. 
+The first part is used if DroneCore is installed locally, and will have to be modified depending on the relative path from your code to the **DroneCore/install/** directory.
+```cmake
+if(WINDOWS)
+    set(dronecore_lib "${CMAKE_SOURCE_DIR}/../../install/lib/dronecore.lib")
+else()
+    set(dronecore_lib "${CMAKE_SOURCE_DIR}/../../install/lib/libdronecore.a")
+endif()
+```
+
+The final section of the file defines the libraries to link against if DroneCore is installed globally.
+```cmake
+target_link_libraries(takeoff_and_land
+    ${dronecore_lib}
+    # If installed system-wide:
+    # dronecore
+    ${CMAKE_THREAD_LIBS_INIT}
+    ${platform_libs}
+)
+```
+
+> **Tip** Getting relative paths right can be difficult. You can use [message()](https://cmake.org/cmake/help/v3.9/command/message.html) to print out debug information in your build:
+```bash
+# Get the location of your CMakeLists.txt file
+message(${CMAKE_SOURCE_DIR})
+```
+
+## Building the App
+
+This section assumes that you have already [built and installed the DroneCore C++ Library](../contributing/build.md) and that your example has a **CMakeLists.txt** in its root directory. 
+
+All we need to do is create a build directory (to keep all our build files in one place) and run the build from there. 
+
+```bash 
+cd /path/to/your/example/  # Open a prompt and navigate to your example 
+mkdir build && cd build   # Create a build directory and navigate into it
+cmake --build .  # Build the current directory
```

Comment:
@julianoes This line was taken from the example docs and didn't work. I tried `cmake ..` and that worked fine. What is "correct"?

Reply:
You mixed Linux and Windows cmake arguments there.

To first, run cmake do:
```
mkdir build && cd build
cmake ..
```

Then the build step on Linux/Mac is (inside this build folder):
```
make -j8
```

And for Windows:
```
cmake --build .
```

Reply:
I don't think you actually need the -j8 bit for Linux do you?

Reply:
No it's just faster.